pageserver/src/http/routes.rs

```diff
@@ -437,6 +437,39 @@ async fn timeline_detail_handler(
     json_response(StatusCode::OK, timeline_info)
 }
 
+async fn walreceiver_detail_handler(
+    request: Request<Body>,
+    _cancel: CancellationToken,
+) -> Result<Response<Body>, ApiError> {
+    let tenant_id: TenantId = parse_request_param(&request, "tenant_id")?;
+    let timeline_id: TimelineId = parse_request_param(&request, "timeline_id")?;
+    check_permission(&request, Some(tenant_id))?;
+
+    let walreceiver_info = async {
+        let tenant = mgr::get_tenant(tenant_id, true).await?;
+
+        let timeline = tenant
+            .get_timeline(timeline_id, false)
+            .map_err(|e| ApiError::NotFound(e.into()))?;
+
+        let walreceiver_status = {
+            match &*timeline.walreceiver.lock().unwrap() {
+                None => "stopping or stopped".to_string(),
+                Some(walreceiver) => match walreceiver.status() {
+                    Some(status) => status.to_human_readable_string(),
+                    None => "Not active".to_string(),
+                },
+            }
+        };
```

Comment:
Feels like this repeats some code we already have in Timeline, perhaps with `Timeline::wait_lsn`. Perhaps we could move the common parts as `Timeline::walreceiver_status(&self) -> String`?

Reply:
Yes, that is good idea. @koivunej 
Will update the code as recommended.